src/Tests/dotnet-workload-install.Tests/GivenFileBasedWorkloadInstall.cs

```diff
@@ -26,6 +27,46 @@ public GivenFileBasedWorkloadInstall(ITestOutputHelper log) : base(log)
             _manifestPath = Path.Combine(_testAssetsManager.GetAndValidateTestProjectDirectory("SampleManifest"), "Sample2.json");
         }
 
+        [Fact]
+        public void InstallStateUpdatesWorkProperly()
+        {
+            (string dotnetRoot, FileBasedInstaller installer, _, _) = GetTestInstaller();
+            var stringFeatureBand = "6.0.300"; // This is hard-coded in the test installer, so if that changes, update this, too.
+            var sdkFeatureBand = new SdkFeatureBand(stringFeatureBand);
+            var path = Path.Combine(dotnetRoot, "metadata", "workloads", stringFeatureBand, "InstallState", "default.json");
+
+            // The install state shouldn't exist, but delete it in case it wasn't cleaned up properly from the last run.
+            if (File.Exists(path))
+            {
+                File.Delete(path);
+            }
```

Comment:
This shouldn't be necessary.  The test infrastructure automatically makes sure that when it creates a test directory, that it's empty.  If the directory already exists, it will first delete it.  If that doesn't work, the test will fail.